lib/src/transports/NativeTransport.dart

```diff
@@ -0,0 +1,114 @@
+import 'dart:convert';
+import 'dart:io';
+
+import '../Logger.dart';
+import '../Message.dart';
+import 'TransportInterface.dart';
+
+final _logger = Logger('Logger::NativeTransport');
+
+class Transport extends TransportInterface {
+  late bool _closed;
+  late String _url;
+  late dynamic _options;
+  WebSocket? _ws;
+
+  Transport(String url, {dynamic options}) : super(url, options: options) {
+    _logger.debug('constructor() [url:$url, options:$options]');
+    this._closed = false;
+    this._url = url;
+    this._options = options ?? {};
+    this._ws = null;
+
+    this._runWebSocket();
+  }
+
+  get closed => _closed;
+
+  @override
+  close() {
+    _logger.debug('close()');
+
+    this._closed = true;
+    this.safeEmit('close');
+
+    try {
+      this._ws?.close();
+    } catch (error) {
+      _logger.error('close() | error closing the WebSocket: $error');
+    }
+  }
+
+  @override
+  Future send(message) async {
+    try {
+      this._ws?.add(jsonEncode(message));
+    } catch (error) {
+      _logger.warn('send() failed:$error');
+    }
+  }
+
+  _onOpen() {
+    _logger.debug('onOpen');
+    this.safeEmit('open');
+  }
+
+  // _onClose(event) {
```

Comment:
Are these comments still needed? Could we remove if the logic is already implemented?

Reply:
I am unsure about this part, this complete function was commented out but I didn't exactly understand myself why it was.